### ISSUE TITLE
False positive for `Lint/UnusedMethodArgument` when passing an error message to `NotImplementedError`

### ISSUE THREAD

**Issue by mishina2228** [rubocop/rubocop#10504]
When passing an error message to `NotImplementedError`,
even if `IgnoreNotImplementedMethods` is set to be true,
RuboCop complains with `Lint/UnusedMethodArgument`.

```ruby
def do_something(unused)
  raise NotImplementedError, 'Not implemented yet'
end
```

--------

## Expected behavior

No offenses detected.

## Actual behavior

```
$ rubocop --debug example.rb
For /tmp: Default configuration from /home/owner/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rubocop-1.26.1/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/owner/.cache/rubocop_cache/5e04273777e1ec6067245135980a93eda5924838/6d7a3b621ca1730e04accd938619e4bdab66cfb1/4099d08316b4e9eec2a24b7d3cc969713274d4be
Inspecting 1 file
Scanning /tmp/example.rb
Loading cache from /home/owner/.cache/rubocop_cache/5e04273777e1ec6067245135980a93eda5924838/6d7a3b621ca1730e04accd938619e4bdab66cfb1/4099d08316b4e9eec2a24b7d3cc969713274d4be
W

Offenses:

example.rb:3:18: W: [Correctable] Lint/UnusedMethodArgument: Unused method argument - unused. If it's necessary, use _ or _unused as an argument name to indicate that it won't be used. You can also write as do_something(*) if you want the method to accept any arguments but don't care about them.
def do_something(unused)
                 ^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.10515446698991582 seconds
```

## Steps to reproduce the problem

Here's a minimal `example.rb` file that shows the problem:
```ruby
# frozen_string_literal: true

def do_something(unused)
  raise NotImplementedError, 'Not implemented yet'
end
```

## RuboCop version

```
$ rubocop -V
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.0 x86_64-linux)
```
